src/sql/parts/objectExplorer/viewlet/serverTreeActionProvider.ts

```diff
@@ -132,7 +132,13 @@ export class ServerTreeActionProvider extends ContributableActionProvider {
 	private getContextKeyService(context: ObjectExplorerContext): IContextKeyService {
 		let scopedContextService = this._contextKeyService.createScoped();
 		let connectionContextKey = new ConnectionContextKey(scopedContextService);
-		connectionContextKey.set(context.profile);
+		let connectionProfile = context.profile;
+		let serverInfo = this._connectionManagementService.getServerInfo(connectionProfile.id);
+		if (serverInfo && serverInfo.serverMajorVersion) {
+			connectionProfile.options  = Object.assign(connectionProfile.options || {},
```

Comment:
I don't think this is quite the right implementation, though it's close. Can we either pass both connectionProfile and serverInfo to the set method, or have a separate context key to handle serverInfo properties?

Reply:
> Can we either pass both connectionProfile and serverInfo to the set method?

`connectionContextKey` implements `IContextKey<IConnectionProfile>` which has `set(value: IConnectionProfile)` method signature. Creating customer`set()` method in `ConnectionContextKey` type to hand over connectionProfile and ServerInfo does not seem to be good solution.

> Can we have a separate context key to handle serverInfo properties?

1) Semantically server information including server version is more likely to be in `ConnectionContextKey` rather than other type of key because server information is properties of connection. We have been putting server information into connection object such as 
`azdata.connection.Connection`, `azdata.IConnectionProfile`, `azdata.ConnectionInfo`

2) It is something that I need spend some time and look into to figure out how to make new ContextKey to be fed to OE in ADS as other keys if you really want to create separate key instead of `ConnectionContextKey`

What is the reason do you think this is not right implementation?